daemon/mgr/image_utils.go

```diff
@@ -0,0 +1,21 @@
+package mgr
+
+import (
+	"regexp"
+	"strings"
+)
+
+// FIXME: need refactor this function.
+// addRegistry add default registry if needed.
+func (mgr *ImageManager) addRegistry(input string) string {
+	if strings.Contains(input, "/") || isNumericID(input) {
```

Comment:
To be honest, I am afraid we can improve this part. Let me introduce one example, pouch pull allencloud/mysql; if I set the registry to be `12.12.12.12:4567` , then I suppose to pull `12.12.12.12:4567/allencloud/mysql`. But code here would ignore this, I think. @zeppp

Reply:
This part shoule can be improved indeed, i agree. But, in fact, when you pull `allencloud/mysql`, it will not add a registry because the input contains "/". So, you will get  `allencloud/mysql`. Only when you pull image by RepoName like `mysql`, it will add the registry you set.

Reply:
> in fact, when you pull allencloud/mysql, it will not add a registry because the input contains "/". So, you will get allencloud/mysql

My point is that `allencloud/mysql` does not include a default registry, if we add a function `addRegistry`, maybe we should add default registry for `allencloud/mysql`, right?

If you agree with this, this is a part of work we need to finish.

Reply:
Oh, I get it. This function `addRegistry` need be refactored while I have no idea about that. @allencloud

Reply:
@zeppp , could we follow the routine like this?

```
if isNumberID(ref) {
   return ref
}

host, remainder := splitDomainHost(ref)
if host.PrefixContain(registry) {
     return ref
} else {
     return registry + ":" + ref
}
```

I checked the oci-image-spec and found that there is no definition about host....

Maybe we need to make rule in `splitDomainHost` to define what is host.....

how do you think? 

cc @allencloud

Reply:
It is a better way. I will update soon. @0x04C2

Reply:
> I checked the oci-image-spec and found that there is no definition about host....

@0x04C2 @zeppp I agree with that we should split the host thing. In addition, could we refactor reference package to define:

``` go
type Ref struct{
        Registry string
        Namespace string
        ImageName string
        Tag string
}
```

for example:

| input      |    registry | namespace  | name| tag |
| :--------: | :--------:| :--: |:--: |:--: |
| docker.io/library/nginx:alpine | docker.io |  library  |nginx| latest|
| localhost:80/nginx:alpine     |   localhost:80 |    |nginx|alpine|
|mysql|||mysql|latest|

If we make it, in every place we can use this `Ref`, and we use the same standard definition when communicating in different modules.

Reply:
If that, could you help to refactor pkg reference? @0x04C2

Reply:
Got it. will do